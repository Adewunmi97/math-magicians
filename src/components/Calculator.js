/* eslint-disable react/prefer-stateless-function */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import View from './View';
import UI from './UI';
import styles from '../assets/calculator.module.css';

class Calculator extends Component {
  render() {
    return (
      <div className={`${styles.Calculator}`}>
        <View output={0} />
        <UI />
      </div>
    );
  }
}

export default Calculator;
