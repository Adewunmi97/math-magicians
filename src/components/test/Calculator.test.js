import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../Calculator';

describe('Test Button', () => {
  it('', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
