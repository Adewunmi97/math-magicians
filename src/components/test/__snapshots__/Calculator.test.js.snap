// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button  1`] = `
<div>
  <div
    class="Calculator"
  >
    <div
      class="View"
      data-testid="screen"
    >
      <p />
      <p />
      <p>
        0
      </p>
    </div>
    <div
      class="UI"
    >
      <div
        class="group"
      >
        <button
          class="Button"
          type="button"
        >
          AC
        </button>
        <button
          class="Button"
          type="button"
        >
          +/-
        </button>
        <button
          class="Button"
          type="button"
        >
          %
        </button>
        <button
          class="Button ButtonOrange"
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="Button"
          type="button"
        >
          8
        </button>
        <button
          class="Button"
          type="button"
        >
          7
        </button>
        <button
          class="Button"
          type="button"
        >
          9
        </button>
        <button
          class="Button ButtonOrange"
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="Button"
          type="button"
        >
          4
        </button>
        <button
          class="Button"
          type="button"
        >
          5
        </button>
        <button
          class="Button"
          type="button"
        >
          6
        </button>
        <button
          class="Button ButtonOrange"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="Button"
          type="button"
        >
          1
        </button>
        <button
          class="Button"
          type="button"
        >
          2
        </button>
        <button
          class="Button"
          type="button"
        >
          3
        </button>
        <button
          class="Button ButtonOrange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="Button ButtonSpan2"
          type="button"
        >
          0
        </button>
        <button
          class="Button"
          type="button"
        >
          .
        </button>
        <button
          class="Button ButtonOrange"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
